dali/python/nvidia/dali/backend_impl.pyi

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, List, overload
+
+from nvidia.dali.data_node import DataNode
+from nvidia.dali.types import DALIDataType
+
+
+class TensorCPU:
+    # @overload
+    # def __init__(self, object: capsule, layout: str = ...) -> None: ...
+    @overload
+    def __init__(self, object, layout: str = ...) -> None:
+        ...
+
+    # @overload
+    # def __init__(self, b: buffer, layout: str = ..., is_pinned: bool = ...) -> None: ...
+    @overload
+    def __init__(self, b, layout: str = ..., is_pinned: bool = ...) -> None:
+        ...
+
+    def _as_gpu(self, *args, **kwargs) -> Any:
+        ...
+
+    # def _expose_dlpack_capsule(self) -> capsule: ...
+    def _expose_dlpack_capsule(self) -> Any:
+        ...
```

Comment:
It starts with underscore, so it's apparently private. Do we want to document that?

Reply:
Good point, I planned to remove the private ones from here, and forgot to do that.